src/libraries/System.Reflection.Context/src/ILLink/ILLink.Suppressions.xml

```diff
@@ -523,5 +523,53 @@
       <property name="Scope">member</property>
       <property name="Target">M:System.Reflection.Context.Delegation.DelegatingType.GetInterface(System.String,System.Boolean)</property>
     </attribute>
+    <attribute fullname="System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute">
```

Comment:
Now that the linker recognizes interfaces on all base types and recursive interfaces, these methods are recognized as overrides of IReflect, which are marked with DAM. There are already suppressions for other missing DAM annotations, so I decided to just suppress the warnings.

Reply:
Yeah, that makes sense to me. But them should we warn if both have an interfaceimpl of I but Derived doesn't override M()? I lean towards yes in this case.

```C#
class Base : I
{
    public virtual void M() {} // warning IL2046
}

class Derived : Base, I // Warning IL2046 pointing to Base.M()?
{ }

interface I
{
    [RequiresUnreferencedCode("RUC")]
    void M();
}
```

Also, in IL it is valid for an abstract Base to implement I but provide no methods. I assume we should still warn on Derived.M() in this case.

```C#
abstract class Base : I
{
}

class Derived : Base
{
    public virtual void M() {} // warning IL2046
}

interface I
{
    [RequiresUnreferencedCode("RUC")]
    void M();
}
```

Reply:
Implemented the simple warning behavior and filed https://github.com/dotnet/runtime/issues/106552